### PR TITLE
python3Packages.python-yate: init at 0.4.1

### PR DESCRIPTION
**Pull request by clerie**
## Description of changes

Release notes: https://github.com/eventphone/python-yate/releases/tag/v0.4.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by NetaliDev — APPROVED** [review]
Result of `nixpkgs-review pr 254410` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.python-yate</li>
    <li>python310Packages.python-yate.dist</li>
    <li>python311Packages.python-yate</li>
    <li>python311Packages.python-yate.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-yate/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+, async-timeout
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "python-yate";
+  version = "0.4.1";
+
```
Comment (NickCao) [inline pkgs/development/python-modules/python-yate/default.nix:12]: Please set `format = "setuptools";`

File: pkgs/development/python-modules/python-yate/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  src = fetchFromGitHub {
+    owner = "eventphone";
+    repo = "python-yate";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-AdnlNsEOFuzuGTBmfV9zKyv2iFHEJ4eLMrC6SHHf7m0=";
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
```
Comment (NickCao) [inline pkgs/development/python-modules/python-yate/default.nix:21]: `aiohttp` does not seem to be a required depencency? https://github.com/eventphone/python-yate/blob/e58fb6f083386e9089e977f404f2e466c3e6cd4f/setup.py#L29

Comment (fabaff) [reply pkgs/development/python-modules/python-yate/default.nix:21]: Looks like that upstream [missed to mention](https://github.com/eventphone/python-yate/blob/master/yate/callgen.py#L7) it.

Comment (clerie) [reply pkgs/development/python-modules/python-yate/default.nix:21]: Yes, this is true

**Review by NickCao — APPROVED** [review]
diff LGTM, waiting for ofborg

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 254410` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.python-yate</li>
    <li>python310Packages.python-yate.dist</li>
    <li>python311Packages.python-yate</li>
    <li>python311Packages.python-yate.dist</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**